templates/skeleton/app/routes/account.addresses.tsx

```diff
@@ -83,119 +100,174 @@ export async function action({request, context}: ActionFunctionArgs) {
       case 'POST': {
         // handle new address creation
         try {
-          const {customerAddressCreate} = await storefront.mutate(
+          const {data, errors} = await customerAccount.mutate(
             CREATE_ADDRESS_MUTATION,
             {
-              variables: {customerAccessToken: accessToken, address},
+              variables: {address, defaultAddress},
             },
           );
 
-          if (customerAddressCreate?.customerUserErrors?.length) {
-            const error = customerAddressCreate.customerUserErrors[0];
-            throw new Error(error.message);
+          if (errors?.length) {
+            throw new Error(errors[0].message);
```

Comment:
There's a chance that throwing could also clear the session, because throwing causes an error page to get rendered, but I don't think that page will commit the session, which might have been updated from line 70. But maybe it's unlikely enough to not worry about it for now?